### ISSUE TITLE
aws_ec2_transit_gateway_vpc_attachment error when auto_accept_shared_attachments = disable

### ISSUE THREAD

**Issue by AzCii** [terraform-providers/terraform-provider-aws#7417]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.11
+ provider.aws v1.57.0

### Affected Resource(s)

* aws_ec2_transit_gateway_vpc_attachment

### Terraform Configuration Files

```hcl
resource "aws_ec2_transit_gateway_vpc_attachment" "test" {
    transit_gateway_id = "tgw-REMOVED"
    
    subnet_ids = ["${module.vpc.private_subnets}"]
    vpc_id = "${module.vpc.vpc_id}"
}
```

### Error

Error: Error applying plan:

1 error(s) occurred:

* aws_ec2_transit_gateway_vpc_attachment.test: 1 error(s) occurred:

* aws_ec2_transit_gateway_vpc_attachment.test: error waiting for EC2 Transit Gateway VPC Attachment (tgw-attach-REMOVED) availability: unexpected state 'pendingAcceptance', wanted target 'available'. last error: %!s(<nil>)

### Expected Behavior

When creating a gateway attachment for a Transit Gateway, shared from a different account, i expect the attachment to be created without warnings, even if auto_accept_shared_attachments is disabled on the Transit Gateway.

### Actual Behavior

When running terraform apply, i get the error above, everything is created as expected, but error is presented anyway.
When doing changes to the attachment i get the same error.
No errors when destroying it.

### Steps to Reproduce

Have Transit Gateway shared with your account via Resource Access management.
Gateway is configured as descriped below

On your account
1. `terraform apply`

### Shared Transit Gateway configuration

Transit Gateway ID: tgw-REMOVED
Owner account ID : REMOVED (shared)
State: available 
Amazon ASN: 64512
DNS support: enable
VPN ECMP support: enable
Auto accept shared attachments: disable
Default association route table: disable
Association route table ID: -
Default propagation route table: disable
Propagation route table ID: -

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 1.59.0 of the Terraform AWS Provider, likely middle of this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.59.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1590-february-14-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
